File: C/selectionSort.c
#include<stdio.h>
#include<stdlib.h>

void main(){
    int n;
    printf("\n----------SELECTION SORT----------\n\n");
    printf("Enter array size: ");
    scanf("%d",&n);
    int *array;
    array = (int*)malloc(sizeof(int)*n);
    printf("Insert array elements: ");

    for(int i=0;i<n;i++){
        scanf("%d",&array[i]);
    }

    int i,j,temp,min;
    //selection sort
    for(i=0;i<n-1;i++){
        min = i;
        for(j=i+1;j<n;j++){
            if(array[j] < array[min])
                min = j;
        }
        //swap
        temp = array[i] + array[min];
        array[min] = temp - array[min];
        array[i] = temp - array[i];
    }
    
    printf("Sorted array: ");
    for(int i=0;i<n;i++){
        printf("%d ",array[i]);
    }
    printf("\n");
    printf("\n--------------EXIT--------------\n");
}

File: Python/selectionSort.py
#Python program to perform selection sort
def selectionSort(array):
    n = len(array)

    for i in range(n-1):
        min = i
        for j in range(i+1,n):
            if(array[j] < array[min]):
                min = j
        array[i],array[min] = array[min],array[i]


if __name__ == "__main__":
    print("\n------SELECTION SORT------\n")
    n = int(input("Enter array size: "))

    array = []

    print("Insert array elements: ")
    for i in range(n):
        ele = int(input())
        array.append(ele)

    selectionSort(array)

    print("Sorted array: ")
    print(array)